Samples/Desktop/D3D12Raytracing/src/D3D12RaytracingSakuraScene/D3D12RaytracingSakuraScene.cpp

```diff
@@ -0,0 +1,1854 @@
+    // In this sample, we only use triangle geometry with a closest hit shader, so others are not set.
+    auto floorHitGroup = raytracingPipeline.CreateSubobject<CD3DX12_HIT_GROUP_SUBOBJECT>();
+    floorHitGroup->SetClosestHitShaderImport(c_floorClosestHitShaderName);
+    floorHitGroup->SetHitGroupExport(c_floorHitGroupName);
+    floorHitGroup->SetHitGroupType(D3D12_HIT_GROUP_TYPE_TRIANGLES);
+
+    auto trunkHitGroup = raytracingPipeline.CreateSubobject<CD3DX12_HIT_GROUP_SUBOBJECT>();
+    trunkHitGroup->SetClosestHitShaderImport(c_trunkClosestHitShaderName);
+    trunkHitGroup->SetHitGroupExport(c_trunkHitGroupName);
+    trunkHitGroup->SetHitGroupType(D3D12_HIT_GROUP_TYPE_TRIANGLES);
+
+    auto leavesHitGroup = raytracingPipeline.CreateSubobject<CD3DX12_HIT_GROUP_SUBOBJECT>();
+    leavesHitGroup->SetClosestHitShaderImport(c_leavesClosestHitShaderName);
+    leavesHitGroup->SetHitGroupExport(c_leavesHitGroupName);
+    leavesHitGroup->SetHitGroupType(D3D12_HIT_GROUP_TYPE_TRIANGLES);
+
+    auto bushHitGroup = raytracingPipeline.CreateSubobject<CD3DX12_HIT_GROUP_SUBOBJECT>();
+    bushHitGroup->SetClosestHitShaderImport(c_bushClosestHitShaderName);
+    bushHitGroup->SetHitGroupExport(c_bushHitGroupName);
+    bushHitGroup->SetHitGroupType(D3D12_HIT_GROUP_TYPE_TRIANGLES);
+
+    auto tcubeHitGroup = raytracingPipeline.CreateSubobject<CD3DX12_HIT_GROUP_SUBOBJECT>();
+    tcubeHitGroup->SetClosestHitShaderImport(c_tcubeClosestHitShaderName);
+    tcubeHitGroup->SetHitGroupExport(c_transparentCubeHitGroupName);
+    tcubeHitGroup->SetHitGroupType(D3D12_HIT_GROUP_TYPE_TRIANGLES);
+
+    // Shader config
+   //  Defines the maximum sizes in bytes for the ray payload and attribute structure.
+    auto shaderConfig = raytracingPipeline.CreateSubobject<CD3DX12_RAYTRACING_SHADER_CONFIG_SUBOBJECT>();
+	UINT payloadSize = sizeof(XMFLOAT4) + sizeof(UINT) + sizeof(UINT); // Since payload is a float4 color, an UINT for recursion depth, and an UINT for reflectHint.
+
+    UINT attributeSize = sizeof(XMFLOAT2);  // float2 barycentrics
+    shaderConfig->Config(payloadSize, attributeSize);
+
+    // Local root signature and shader association
+    // This is a root signature that enables a shader to have unique arguments that come from shader tables.
+    CreateLocalRootSignatureSubobjects(&raytracingPipeline);
+
+    // Global root signature
+    // This is a root signature that is shared across all raytracing shaders invoked during a DispatchRays() call.
+    auto globalRootSignature = raytracingPipeline.CreateSubobject<CD3DX12_GLOBAL_ROOT_SIGNATURE_SUBOBJECT>();
+    globalRootSignature->SetRootSignature(m_raytracingGlobalRootSignature.Get());
+
+    // Pipeline config
+    // Defines the maximum TraceRay() recursion depth.
+    auto pipelineConfig = raytracingPipeline.CreateSubobject<CD3DX12_RAYTRACING_PIPELINE_CONFIG_SUBOBJECT>();
+    UINT maxRecursionDepth = 9;
```

Comment:
Don't need this to be so high.  Might as well be 2 at most given the nature of the scene.  Be sure to make trace logic also stop at the updated depth limit.